source/YTEPs/YTEP-0037.rst

```diff
@@ -0,0 +1,261 @@
+
+so it is expected that ``black`` plays nicely with ``flake8`` by construction.
+``black`` applies an opinated style, and offers very little configuration options
+by design. Only the target line length can be changed. This makes it a critical 
+point, requiring discussion if this YTEP is approved.
+
+**maximal line length**
+
+The guidelines states that
+
+    Line widths should not be more than 80 characters
+
+Despite this being respected in most of the code base, there remains a large amount of
+outliers, that would be time-consuming to go through by hand. Taking the example of the
+yt-4.0 branch at the time of writing, there are 2158 lines exceeding 80 characters
+(~1.5% of the whole code base), or, visually
+
+.. image:: ../images/lw-ytep-37.png
+
+Note that when long strings are present, ``black`` will not attempt to split them to
+shorten the individual lines.
+This is most important in the case of dosctrings, and I explore the tools available to
+validate them hereafter (see `additional rules`_).
+
+There is a range of possible values we might give preference to. Python's standard
+library caps linelenght at 79, pandas does so at 80. By default, ``black`` will target
+88, as its authors claim it reduces the total number of lines by some 10% (as compared
+to enforcing 80).
+
+In first drafting the PR linked above, I chose a line-lenght of 100, so as to minimize
+the amount of manual tweaking left to me after a ``black`` pass. I estimated that
+imposing a strict limit to 80 chars would leave 545 lines to be manually updated, while
+caping at 88 leaves a mere 135 (75% less work). As a reference, ``dask`` uses black's
+default settings, *and* allows flexibility for docstrings up to 120 characters through
+flake8.
+
+
+**sorting imports**
+
+PEP8 `recommends sorting imports statments <https://www.python.org/dev/peps/pep-0008/#imports>`_,
+Needless to say, the task is daunting and definitely not worthy of anyone's time if we
+had to go back and apply those rules manually to the code base.
+Luckilly, ``isort`` is able to check for and auto-apply those rules, so it can easily be
+added to the CI-linting process.
+
+Moreover, ``isort`` is configurable so that it allows for the definition of custom
+"sections" within import statements. This can be use to isolate imports from ``unyt``,
+which falls somewhere in between the default sections "third party" (external
+dependency) and "first party" (the project). This is done in 
+`#2592 <https://github.com/yt-project/yt/pull/2592>`_.
+
+To better highlight the way yt 4.0 depends on ``unyt``, I also propose that, within the
+code base, we import directly from ``unyt`` as often as possible, so as to limit
+confusion between ``unyt`` and its wrapper interface ``yt.units``. This is done in 
+`#2597 <https://github.com/yt-project/yt/pull/2597>`_.
+
+.. _additional rules:
+**additional rules & flake8 pluggings**
+
+Since the oldest python version supported (as of yt-4.0) is 3.6, it means we can start
```

Comment:
`setup.py` indicates that we still support Python 3.5. I don't have much problem increasing that to 3.6, but I think this warrants a discussion about whether it's better to do this pre or post releasing yt-4.0.

Reply:
You're absolutely right, but there is room for interpretation here. We don't run tests for any version older that the latest 3.6, hence my phrasing.
The fstring experiment I'm running in https://github.com/yt-project/yt/pull/2605 has some flaws anyway, since the tool (flynt) is fairly new.
This part of the YTEP can absolutely be postponed.